Tests/Functional/Asynchronous/Notifications/Expectations/main.swift

```diff
@@ -0,0 +1,78 @@
+// CHECK: .*/Tests/Functional/Asynchronous/Notifications/Expectations/main.swift:49: error: NotificationExpectationsTestCase.test_observeNotificationWithIncorrectName_fails : Asynchronous wait failed - Exceeded timeout of 0.1 seconds, with unfulfilled expectations: Expect notification 'expectedName' from any object
+// CHECK: Test Case 'NotificationExpectationsTestCase.test_observeNotificationWithIncorrectName_fails' failed \(\d+\.\d+ seconds\).
+    func test_observeNotificationWithIncorrectName_fails() {
+        expectationForNotification("expectedName", object: nil, handler:nil)
+        NSNotificationCenter.defaultCenter().postNotificationName("actualName", object: nil)
+        waitForExpectationsWithTimeout(0.1, handler: nil)
+    }
+    
+// CHECK: Test Case 'NotificationExpectationsTestCase.test_observeNotificationWithIncorrectObject_fails' started.
+// CHECK: .*/Tests/Functional/Asynchronous/Notifications/Expectations/main.swift:61: error: NotificationExpectationsTestCase.test_observeNotificationWithIncorrectObject_fails : Asynchronous wait failed - Exceeded timeout of 0.1 seconds, with unfulfilled expectations: Expect notification 'notificationWithIncorrectObjectTest' from dummyObject
+// CHECK: Test Case 'NotificationExpectationsTestCase.test_observeNotificationWithIncorrectObject_fails' failed \(\d+\.\d+ seconds\).
+    func test_observeNotificationWithIncorrectObject_fails() {
+        let notificationName = "notificationWithIncorrectObjectTest"
+        let dummyObject: NSString = "dummyObject"
+        let anotherDummyObject = NSObject()
+        expectationForNotification(notificationName, object: dummyObject, handler: nil)
+        NSNotificationCenter.defaultCenter().postNotificationName(notificationName, object:anotherDummyObject)
+        waitForExpectationsWithTimeout(0.1, handler: nil)
+    }
+    
+    static var allTests: [(String, NotificationExpectationsTestCase -> () throws -> Void)] {
+        return [
+                   ("test_observeNotificationWithName_passes", test_observeNotificationWithName_passes),
+                   ("test_observeNotificationWithNameAndObject_passes", test_observeNotificationWithNameAndObject_passes),
+                   ("test_observeNotificationWithNameAndObject_butExpectingNoObject_passes", test_observeNotificationWithNameAndObject_butExpectingNoObject_passes),
+                   ("test_observeNotificationWithIncorrectName_fails", test_observeNotificationWithIncorrectName_fails),
+                   ("test_observeNotificationWithIncorrectObject_fails", test_observeNotificationWithIncorrectObject_fails),
+        ]
+    }
+}
+
+XCTMain([testCase(NotificationExpectationsTestCase.allTests)])
+
+// CHECK: Executed 5 tests, with 2 failures \(0 unexpected\) in \d+\.\d+ \(\d+\.\d+\) seconds
+// CHECK: Total executed 5 tests, with 2 failures \(0 unexpected\) in \d+\.\d+ \(\d+\.\d+\) seconds
```

Comment:
nit-pick: Add a new line at the end of this file so Git won't complain :stuck_out_tongue: